virtool/uploads/api.py

```diff
@@ -168,10 +158,9 @@ async def delete(self) -> Union[r204, r401, r403, r404]:
             403: Not permitted
             404: Not found
         """
-        pg = self.request.app["pg"]
         upload_id = int(self.request.match_info["id"])
 
-        upload = await virtool.uploads.db.delete(self.request, pg, upload_id)
+        upload = await get_data_from_req(self.request).uploads.delete(upload_id)
 
         if not upload:
             raise NotFound()
```

Comment:
Use `ResourceNotFoundError` pattern instead. See already implemented pieces of the data layer.

Do this for all other applicable situations in this module.